Decode received byte[] and encode message to send

charlesrollin opened the pull request.

Commit:
Decode received byte[] and encode message to send

melaniebrd commented on src/fr/mrc/ptichat/main/processing/RequestHandler.java:
```diff
@@ -0,0 +1,43 @@
+     * @return <code>true</code> if the array represents a message, false otherwise
+     */
+    public boolean isMessage(byte[] input) {
+        return this.checkFirstByte(input, 'M');
```
@CRollin : Est-ce que ce ne serait pas mieux de définir et utiliser une constante pour `'M'` ? En cas de changement de la lettre, il y aura moins de code a changer

charlesrollin replied:
@melaniebrd: Indeed! Should we define this in an external class?

melaniebrd commented on src/fr/mrc/ptichat/main/processing/RequestHandler.java:
```diff
@@ -0,0 +1,43 @@
+     * @return <code>true</code> if the array represents a termination signal, false otherwise
+     */
+    public boolean isTerminationBytes(byte[] input) {
+        return this.checkFirstByte(input, 'Q');
```
Same for `(input, 'Q');`

melaniebrd commented on src/fr/mrc/ptichat/main/processing/RequestHandler.java:
```diff
@@ -0,0 +1,43 @@
+     */
+    public String getMessage(byte[] input) {
+        if (!this.isMessage(input)) throw new IllegalArgumentException();
+        return new String(Arrays.copyOfRange(input, 1, input.length));
```
@CRollin par curiosité, que représente le 1 dans `Arrays.copyOfRange(input, 1, input.length)` ?

charlesrollin replied:
We are only interested in input minus its first element, hence in indexes from 1 (included) to input.length (excluded).

melaniebrd commented on src/fr/mrc/ptichat/test/RequestHandlerTest.java:
```diff
@@ -0,0 +1,52 @@
+    public void isTerminationBytes() throws Exception {
+        assertFalse(rh.isTerminationBytes(this.EMPTY_MSG_INPUT));
+        assertFalse(rh.isTerminationBytes(this.MSG_INPUT));
+
```
@CRollin est-ce qu'il y a une raison particulière pour laquelle tu sautes des lignes entre 

```
assertTrue(rh.isMessage(this.EMPTY_MSG_INPUT));
assertTrue(rh.isMessage(this.MSG_INPUT));

assertFalse(rh.isMessage(null));
assertFalse(rh.isMessage(this.TERMINATION_INPUT));
```

Et 

```
assertFalse(rh.isTerminationBytes(this.EMPTY_MSG_INPUT));
assertFalse(rh.isTerminationBytes(this.MSG_INPUT));

assertFalse(rh.isTerminationBytes(null));
assertTrue(rh.isTerminationBytes(this.TERMINATION_INPUT));
```

?

melaniebrd commented on src/fr/mrc/ptichat/test/RequestHandlerTest.java:
```diff
@@ -0,0 +1,52 @@
+        }
+    }
+
+}
```
`no newline at end of file`je sais pas s'il faut en mettre ou pas :p

melaniebrd reviewed (commented):
J'ai été un peu relou sur certains trucs désolée <3

Commit:
Move constants for 1st byte of a message to .properties file

A specific class now gives access to the keys stored in
the .properties file. This will ease changes in the code
if these values are modified in the future.

melaniebrd reviewed (approved):
Sounds good to me :) Approved
